packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -1,524 +1,25 @@
-            const inits = [];
-            privateMapInits.push(inits);
-
-            privateMaps.push(
-              buildClassPrivateProperty(t.thisExpression(), prop, inits, state),
-            );
-          }
-        }
-        let p = 0;
-        for (const prop of props) {
-          if (prop.node.static) {
-            if (prop.isPrivate()) {
-              staticNodes.push(
-                ...buildClassStaticPrivateProperty(
-                  t.cloneNode(ref),
-                  prop,
-                  state,
-                ),
-              );
-            } else {
-              staticNodes.push(
-                buildClassProperty(t.cloneNode(ref), prop, state),
-              );
-            }
-          } else if (prop.isPrivate()) {
-            instanceBody.push(privateMaps[p]());
-            staticNodes.push(...privateMapInits[p]);
-            p++;
-          } else {
-            instanceBody.push(
-              buildClassProperty(t.thisExpression(), prop, state),
-            );
-          }
-        }
-
-        if (instanceBody.length) {
-          if (!constructor) {
-            const newConstructor = t.classMethod(
-              "constructor",
-              t.identifier("constructor"),
-              [],
-              t.blockStatement([]),
-            );
-            if (isDerived) {
-              newConstructor.params = [t.restElement(t.identifier("args"))];
-              newConstructor.body.body.push(
-                t.expressionStatement(
-                  t.callExpression(t.super(), [
-                    t.spreadElement(t.identifier("args")),
-                  ]),
-                ),
-              );
-            }
-            [constructor] = body.unshiftContainer("body", newConstructor);
-          }
-
-          const state = { scope: constructor.scope };
-          for (const prop of props) {
-            if (prop.node.static) continue;
-            prop.traverse(referenceVisitor, state);
-          }
-
-          //
-
-          if (isDerived) {
-            const bareSupers = [];
-            constructor.traverse(findBareSupers, bareSupers);
-            for (const bareSuper of bareSupers) {
-              bareSuper.insertAfter(instanceBody);
-            }
-          } else {
-            constructor.get("body").unshiftContainer("body", instanceBody);
-          }
-        }
-
-        for (const prop of props) {
-          prop.remove();
-        }
-
-        if (computedNodes.length === 0 && staticNodes.length === 0) return;
-
-        if (path.isClassExpression()) {
-          path.scope.push({ id: ref });
-          path.replaceWith(
-            t.assignmentExpression("=", t.cloneNode(ref), path.node),
-          );
-        } else if (!path.node.id) {
-          // Anonymous class declaration
-          path.node.id = ref;
-        }
-
-        path.insertBefore(computedNodes);
-        path.insertAfter(staticNodes);
-      },
+    manipulateOptions(opts, parserOpts) {
```

Comment:
hm, it's already done in `classFeatures`, so I would expect this to either override classFeatures hook, or duplicate those entries in plugins array, which one is it? does this hook need to be even here if its already specified by classFeatures?

Reply:
When we are using the `@babel/plugin-propsal-class-properties` plugins, `@babel/plugin-class-features` is called without the `fields` option. Thus, it doesn' enable any parser plugin and it needs to be done here.